src/components/Reports/ReportsTrafficSources/ReportsTrafficSources.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import { InjectAppServices } from '../../../services/pure-di';
+import { FormattedMessage, injectIntl } from 'react-intl';
+import Loading from '../../Loading/Loading';
+
+const ReportsTrafficSources = function({
+  domainName,
+  dateFrom,
+  intl,
+  dependencies: { datahubClient },
+}) {
+  const [trafficSources, setTrafficSources] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [totalQuantity, setTotalQuantity] = useState(0);
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setIsLoading(true);
+      const trafficSourcesData = await datahubClient.getTrafficSourcesByPeriod(
+        domainName,
+        dateFrom,
+      );
+      setTrafficSources(trafficSourcesData);
+      let total = 0;
+      trafficSources.forEach((item, index) => {
+        total += item.quantity;
+      });
+      setTotalQuantity(total);
+      setIsLoading(false);
+    };
+
+    fetchData();
+  }, [datahubClient, dateFrom, domainName, trafficSources]);
```

Comment:
it is weird using `trafficSources` as a dependency, it is the result. Am I missing something?

Reply:
⚠️

Reply:
Line 34:  React Hook useEffect has a missing dependency: 'trafficSources'. Either include it or remove the dependency array  react-hooks/exhaustive-deps

Reply:
Ok, that is a warning, that does not mean that the current solution is fine.

Reply:
If you cannot deal with this, please call me.

Reply:
Also, it would be better if we change the state only a time, maybe we should have complex objects in place of primitives, because all of them are changed together, let me know if I cannot explain myself.

Reply:
yes i changed